File: frontend/src/app/models/domain/course.new.dto.ts

export class CourseNewDTO {

    // FIELDS

    private id: number;
    private name: string;
    private period: string;
    private type: string;

    // CONSTRUCTOR

    constructor() {}

    // GETTERS / SETTERS

    public getId(): number {
        return this.id;
    }

    public setId(id: number) {
        this.id = id;
    }

    public getName(): string {
        return this.name;
    }

    public setName(name: string) {
        this.name = name;
    }

    public getPeriod(): string {
        return this.period;
    }

    public setPeriod(period: string) {
        this.period = period;
    }

    public getType(): string {
        return this.type;
    }

    public setType(type: string) {
        this.type = type;
    }
}

File: backend/src/main/java/com/williamdsw/semsys/domain/dto/CourseNewDTO.java
package com.williamdsw.semsys.domain.dto;

import java.io.Serializable;

import javax.validation.constraints.NotNull;

public class CourseNewDTO implements Serializable
{
	// FIELDS 
	
	private static final long serialVersionUID = 1L;
	private Integer id;
	
	@NotNull(message = "Name is required")
	private String name;
	
	@NotNull(message = "Period is required")
	private String period;
	
	@NotNull(message = "Type is required")
	private String type;
	
	// CONSTRUCTORS
	
	public CourseNewDTO () {}
	
	// GETTERS / SETTERS
	
	public Integer getId () 
	{
		return id;
	}
	
	public void setId (Integer id) 
	{
		this.id = id;
	}
	
	public String getName () 
	{
		return name;
	}
	
	public void setName (String name) 
	{
		this.name = name;
	}
	
	public String getPeriod () 
	{
		return period;
	}
	
	public void setPeriod (String period) 
	{
		this.period = period;
	}
	
	public String getType () 
	{
		return type;
	}
	
	public void setType (String type) 
	{
		this.type = type;
	}
}